src/sql/azdata.d.ts

```diff
@@ -1929,6 +1929,10 @@ declare module 'azdata' {
 		taskId: number;
 	}
 
+	export interface CreateSasResponse {
```

Comment:
These should go into `azdata.proposed.d.ts`

And in general - do you actually need these in the extension API? I guess it could be useful to have this stuff available to extensions, but for your case is this actually necessary? The dialog lives in core so you shouldn't need to have the extension be asking for a SAS. You can just have this be an internal type that's passed across the JSON RPC channel.

Reply:
So `CreateSas` should go in `azdata`? I don't have to change `mssql` then.

Reply:
So mssql will still be the thing that has the code to call CreateSas (since that is done by STS). But you don't need the extension API to do that - look at how I hooked up the Azure code where the JSON RPC calls will call directly into the Azure extension to make the calls there. So you'd just do something similar for the mssql extension. 

https://github.com/microsoft/azuredatastudio/blob/5547c2baba219af17a6fc574d1d5d8eaabe48742/src/sql/workbench/api/common/extHostAzureAccount.ts

The response type would then live in mssql as well, so that wouldn't go here.